File: android/src/main/java/com/revenuecat/purchases/react/RNPaywallManager.java
package com.revenuecat.purchases.react;

import androidx.annotation.NonNull;

import com.facebook.react.uimanager.SimpleViewManager;
import com.facebook.react.uimanager.ThemedReactContext;

public class RNPaywallManager extends SimpleViewManager<Paywall> {
    public static final String REACT_CLASS = "RNPaywall";

    @NonNull
    @Override
    public String getName() {
        return REACT_CLASS;
    }

    @NonNull
    @Override
    protected Paywall createViewInstance(@NonNull ThemedReactContext themedReactContext) {
        return new Paywall(themedReactContext);
    }
}

File: react-native-purchases-ui/android/src/main/java/com/revenuecat/purchases/react/ui/RNPaywallManager.kt
package com.revenuecat.purchases.react.ui

import android.view.View
import com.facebook.react.uimanager.LayoutShadowNode
import com.facebook.react.uimanager.SimpleViewManager
import com.facebook.react.uimanager.ThemedReactContext
import com.facebook.yoga.YogaMeasureFunction
import com.facebook.yoga.YogaMeasureMode
import com.facebook.yoga.YogaMeasureOutput
import com.facebook.yoga.YogaNode
import com.revenuecat.purchases.ui.revenuecatui.ExperimentalPreviewRevenueCatUIPurchasesAPI
import com.revenuecat.purchases.ui.revenuecatui.views.PaywallView

@OptIn(ExperimentalPreviewRevenueCatUIPurchasesAPI::class)
internal class RNPaywallManager : SimpleViewManager<PaywallView>() {
    companion object {
        const val REACT_CLASS = "RNPaywall"
    }

    override fun getName(): String {
        return REACT_CLASS
    }

    override fun createViewInstance(themedReactContext: ThemedReactContext): PaywallView {
        return PaywallView(themedReactContext)
    }

    override fun createShadowNodeInstance(): PaywallViewShadowNode {
        return PaywallViewShadowNode()
    }
}
